### PR TITLE
mgmt resources fix TODOs

### PR DESCRIPTION
**Pull request by xseeseesee**

### CODE REVIEW

File: sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java [hunk 1]
```diff
@@ -115,20 +125,67 @@ public static String createOdataFilterForTags(String tagName, String tagValue) {
      * Download a file asynchronously.
      *
      * @param url the URL pointing to the file
-     * @param retrofit the retrofit client
+     * @param httpPipeline the http pipeline
      * @return an Observable pointing to the content of the file
      */
+    public static Mono<byte[]> downloadFileAsync(String url, HttpPipeline httpPipeline) {
```
Comment (weidongxu-microsoft) [inline sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java:131]: Why it is in main? I assume this is only needed in test?

Comment (xccc-msft) [reply sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java:131]: It's required by [AppServiceCertificateImpl](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/appservice/mgmt/src/main/java/com/azure/resourcemanager/appservice/implementation/AppServiceCertificateImpl.java#L123).

File: sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java [hunk 2]
```diff
@@ -115,20 +123,48 @@ public static String createOdataFilterForTags(String tagName, String tagValue) {
...
+        FileService service = RestProxy.create(FileService.class, httpPipeline);
+        try {
+            return service.download(getHost(url), getPathAndQuery(url))
+                .flatMap(response -> FluxUtil.collectBytesInByteBufferStream(response.getValue()));
+        } catch (MalformedURLException ex) {
+            return Mono.empty();
```
Comment (weidongxu-microsoft) [inline sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java:135]: Maybe a `Mono.error` would be more proper?

File: sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java [hunk 3]
```diff
@@ -115,20 +123,48 @@ public static String createOdataFilterForTags(String tagName, String tagValue) {
...
+     * Get path from url.
+     *
+     * @param urlString the url string
+     * @return the path
+     * @throws MalformedURLException when the url is invalid format
+     */
+    public static String getPathAndQuery(String urlString) throws MalformedURLException {
+        URL url = new URL(urlString);
+        String path = url.getPath();
+        String query = url.getQuery();
```
Comment (ChenTanyi) [inline sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java:163]: I think it should be `getRawPath` and `getRawQuery` due to you just do a string add.

Comment (xccc-msft) [reply sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java:163]: Not available for URL. Such methods are in URI.

Comment (weidongxu-microsoft) [reply sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java:163]: Could use a bit testing (send one to httpbin.org).

**Pull request merged**